lib/strings.nix

```
@@ -355,6 +355,21 @@ rec {
   */
   escapeC = list: replaceStrings list (map (c: "\\x${ toLower (lib.toHexString (charToInt c))}") list);
 
+  /* Escape the string so it can be safely placed inside a URL
+     query.
+
+     Type: escapeQuery = string -> string
+
+     Example:
+       escapeQuery "foo bar"
+       => "foo%20bar"
```

```suggestion
       => "foo%20bar"
       
       escapeQuery "foo/bar"
       => "foo%2Fbar"
```

Make it clear that we are escaping reserved characters, as for some use cases it might make sense to leave them be. If this turns out to be needed in the future we can always make the set of allowed characters a parameter.

---

How would you make it a parameter if you don't leave room for it now?

---

`escapeURLWithAllowedCharsButShorterName`, and define `escapeURL = escapeURLWithAllowedCharsButShorterName []`. I think it's good to have "`escapeURL`" do the obvious scheme-agnostic thing anyway.

---

Updated the example, and also the test cases.